azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -781,3 +790,42 @@ def _remove_secrets_from_secret_sync(
                     resource=secret_sync,
                 )
                 return wait_for_terminal_state(poller)
+
+    def _get_secret_names(self, keyvault_name: str) -> List[str]:
+        secret_iteratable = self.keyvault_client.get_secrets(
+            vault_base_url=KEYVAULT_URL.format(keyvaultName=keyvault_name)
+        )
+        return [secret.get("id") for secret in secret_iteratable]
+
+    def _begin_delete_secret(self, keyvault_name: str, secret_name: str):
+        # Construct vault URL
+        vault_url = KEYVAULT_URL.format(keyvaultName=keyvault_name)
+
+        # Initiate deletion
+        pipeline_response = self.keyvault_client.delete_secret(
+            vault_base_url=vault_url,
+            secret_name=secret_name,
+            cls=lambda pipeline_response, _, __: pipeline_response,
+        )
+
+        for attempt in range(SECRET_DELETE_MAX_RETRIES):
+            try:
+                # Check if secret is deleted
+                self.keyvault_client.get_deleted_secret(
+                    vault_base_url=vault_url,
+                    secret_name=secret_name,
+                )
+                return  # Exit if no exception, deletion confirmed
+            except ResourceNotFoundError:
```

Comment:
response of 404 will raise ResourceNotFoundError over an HttpResponseError ?

Reply:
not sure if I understand your question correctly, according to the reference you shared with me: https://github.com/Azure/azure-sdk-for-python/blob/4567dbfd0236f85c4d9d311a447ef2add99a48cc/sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_shared/_polling.py#L113

My understanding is --- In the case without permission to call get_deleted_* will keep raising ResourceNotFoundError and eventually raise 403 when the requested resource get recovered, at that point the permission issue is identify and we should exit the loop and raise "fail to delete" exception